test/byContract/IexecPoco/IexecPoco1.test.ts

```diff
@@ -408,38 +504,312 @@ describe('IexecPoco1', () => {
+            // Compute prices, stakes, rewards, ...
+            const dealPrice =
+                (appPrice + datasetPrice + workerpoolPrice) * // task price
+                volume;
+            const schedulerStake = await iexecWrapper.computeSchedulerDealStake(
+                workerpoolPrice,
+                volume,
+            );
+            // Deposit required amounts.
+            await iexecWrapper.depositInIexecAccount(sponsor, dealPrice);
+            await iexecWrapper.depositInIexecAccount(scheduler, schedulerStake);
+            // Sign and match orders.
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            const dealId = getDealId(iexecWrapper.getDomain(), orders.requester);
+            const tx = iexecPocoAsSponsor.sponsorMatchOrders(...orders.toArray());
+            // Check balances and frozen.
+            await expect(tx).to.changeTokenBalances(
+                iexecPoco,
+                [iexecPoco, sponsor, scheduler, requester],
+                [dealPrice + schedulerStake, -dealPrice, -schedulerStake, 0],
+            );
+            expect(await iexecPoco.frozenOf(requester.address)).to.equal(0);
+            expect(await iexecPoco.frozenOf(sponsor.address)).to.equal(dealPrice);
+            expect(await iexecPoco.frozenOf(scheduler.address)).to.equal(schedulerStake);
+            // Check events.
+            await expect(tx).to.emit(iexecPoco, 'OrdersMatched');
+            // Check deal
+            const deal = await iexecPoco.viewDeal(dealId);
+            expect(deal.sponsor).to.equal(sponsor.address);
+        });
+
+        it('Should fail when sponsor has insufficient balance', async () => {
+            // Compute prices, stakes, rewards, ...
+            const dealPrice =
+                (appPrice + datasetPrice + workerpoolPrice) * // task price
+                volume;
+            const schedulerStake = await iexecWrapper.computeSchedulerDealStake(
+                workerpoolPrice,
+                volume,
+            );
+            // Deposit less than deal price in the sponsor's account.
+            await iexecWrapper.depositInIexecAccount(sponsor, dealPrice - 1);
+            await iexecWrapper.depositInIexecAccount(scheduler, schedulerStake);
+            // Sign and match orders.
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            await expect(
+                iexecPocoAsSponsor.sponsorMatchOrders(...orders.toArray()),
+            ).to.be.revertedWith('IexecEscrow: Transfer amount exceeds balance');
+        });
     });
-    describe('[TODO] Sponsor match orders', () => {});
 
     /**
      * Helper function to deposit requester and scheduler stakes with
      * default prices for tests that do not rely on price changes.
      */
-    async function depositForRequesterAndSchedulerWithDefaultPrices() {
+    async function depositForRequesterAndSchedulerWithDefaultPrices(volume: number) {
```

Comment:
```suggestion
    async function depositForRequesterAndSchedulerWithDefaultPrices(volume: number = 1) {
```
and use `depositForRequesterAndSchedulerWithDefaultPrices()` ?

Reply:
It's a good idea but the goal here is to use the global constant `volume` and not duplicate its value elsewhere.